### PR TITLE
[ci]Add Ascend NPU native support to the unit test code

### PR DESCRIPTION
**Pull request by Crystal-jiang**
### What does this PR do?

Currently, the unit test code fails to run on NPU platforms due to platform-specific dependencies. This PR refactors the unit test code related to CUDA and NCCL to make it compatible with Ascend NPU and any other third-party platforms in VeOmni. The changes ensure that the test cases can be executed seamlessly across different hardware backends.

The changes in this PR fix the hardcoded GPU-related code in the unit tests, making it hardware-agnostic, which allows pytest to start successfully on the NPU platform. For more details, please refer to issue [Unit Testing: Most Test Cases Failed to Meet Expectations](https://github.com/ByteDance-Seed/VeOmni/issues/215).

NPU results：
image

GPU results：
image

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `misc`, `ci`, `config`, `docs`, `core`, `data`, `dist`, `omni`, `logging`, `model`, `optim`, `ckpt`, `release`, `task`, `
  - If this PR involves multiple modules, separate them with `,` like `[megatron, torch, liger-kernals,fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, torch, liger-kernals] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `make commit && make style && make quality`
- [ ] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: tests/utils/test_trainer_saveload.py [hunk 1]
```diff
@@ -257,11 +258,12 @@ def main():
                 helper.print_example(example=micro_batches[0], rank=args.train.local_rank)
 
             total_loss = 0
-            torch.cuda.synchronize()
+            get_torch_device().synchronize()
```
Comment (onehaitao) [inline tests/utils/test_trainer_saveload.py]: can replace with `synchronize()` function from `veomni/utils/device.py`

File: tests/utils/test_trainer_saveload.py [hunk 2]
```diff
@@ -284,7 +286,7 @@ def main():
 
             # collect mean loss across data parallel group
             total_loss, grad_norm = all_reduce((total_loss, grad_norm), group=get_parallel_state().fsdp_group)
-            torch.cuda.synchronize()
+            get_torch_device().synchronize()
```
Comment (onehaitao) [inline tests/utils/test_trainer_saveload.py]: same as above

File: tests/utils/test_trainer_saveload.py [hunk 3]
```diff
@@ -346,7 +348,7 @@ def step_id(s):
     check_state_dict(golden_model_sd, model.state_dict(), tied_weights_keys)
     check_state_dict(golden_optim_sd, optimizer.state_dict(), need_flatten=True)
 
-    torch.cuda.synchronize()
+    get_torch_device().synchronize()
```
Comment (onehaitao) [inline tests/utils/test_trainer_saveload.py]: same as above

File: tests/utils/test_trainer_saveload.py (hunk 1 above)
Comment (Crystal-jiang) [reply tests/utils/test_trainer_saveload.py]: modified

File: tests/utils/test_trainer_saveload.py (hunk 2 above)
Comment (Crystal-jiang) [reply tests/utils/test_trainer_saveload.py]: modified

File: tests/utils/test_trainer_saveload.py (hunk 3 above)
Comment (Crystal-jiang) [reply tests/utils/test_trainer_saveload.py]: modified

**Comment by FoolPlayer** [comment]
Please try to run pre-commit before 'git commit'.

**Comment by onehaitao** [comment]
label ascend
